src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -858,6 +858,19 @@ private bool ParseFile(string[] args, ref int i, bool noexitSeen)
             // of the script to evaluate. If -file comes before -command, it will
             // treat -command as an argument to the script...
 
+            bool? GetBoolValue(string arg)
+            {
+                if (arg.Equals("$true", StringComparison.OrdinalIgnoreCase))
```

Comment:
I think this needs to be `True`, not `$true`.

Or maybe it needs to be both `$true` and `True` - because it depends on the shell how `$true` is expanded.